v3/lgr_hyper_ep.hpp

```diff
@@ -0,0 +1,648 @@
+#ifndef LGR_HYPER_EP_HPP
+#define LGR_HYPER_EP_HPP
+
+//#define LGR_COMPILE_TIME_MATERIAL_BRANCHES
```

Comment:
@ibaned, this section needs to be changed for compile time material branching.  Below, if `LGR_COMPILE_TIME_MATERIAL_BRANCHES` is defined, we include a user header file that gives user inputs and define `IF_MAT_PROS_*` to use `if constexpr` to branch based on the inputs in that header file.  If it is not defined, runtime branching is used.

There is still runtime branching, but every branch that can be known at the time of parsing user input is constructed in this way.

What do you envision the templated branching look like?

Reply:
```cpp
template <class HardeningLaw>
void material_model(double input1, double& output1) {
  HardeningLaw::compute(input1, output1);
}
```